From ecf6f8b2371d56bc6454d6775c8a5424607257dc Mon Sep 17 00:00:00 2001
From: Myy <myy@miouyouyou.fr>
Date: Wed, 18 Jan 2017 00:07:55 +0000
Subject: [PATCH] Adaptation of Shawn Lin's patch muting the MMC driver

This avoid spam from the MMC driver in the kernel logs. However, I'm
wondering if that doesn't hide a real configuration problem...

Original patch URI:
http://lists.infradead.org/pipermail/linux-rockchip/2017-January/013621.html

Signed-off-by: Myy <myy@miouyouyou.fr>
---
 drivers/mmc/host/dw_mmc.c | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/drivers/mmc/host/dw_mmc.c b/drivers/mmc/host/dw_mmc.c
index 192e586..2939421 100644
--- a/drivers/mmc/host/dw_mmc.c
+++ b/drivers/mmc/host/dw_mmc.c
@@ -1179,11 +1179,12 @@ static void dw_mci_setup_bus(struct dw_mci_slot *slot, bool force_clkinit)
 		if ((clock != slot->__clk_old &&
 			!test_bit(DW_MMC_CARD_NEEDS_POLL, &slot->flags)) ||
 			force_clkinit) {
-			dev_info(&slot->mmc->class_dev,
-				 "Bus speed (slot %d) = %dHz (slot req %dHz, actual %dHZ div = %d)\n",
-				 slot->id, host->bus_hz, clock,
-				 div ? ((host->bus_hz / div) >> 1) :
-				 host->bus_hz, div);
+			if (!force_clkinit)
+				dev_info(&slot->mmc->class_dev,
+					"Bus speed (slot %d) = %dHz (slot req %dHz, actual %dHZ div = %d)\n",
+					slot->id, host->bus_hz, clock,
+					div ? ((host->bus_hz / div) >> 1) :
+					host->bus_hz, div);
 
 			/*
 			 * If card is polling, display the message only
@@ -3356,8 +3357,7 @@ int dw_mci_runtime_resume(struct device *dev)
 			continue;
 		if (slot->mmc->pm_flags & MMC_PM_KEEP_POWER)
 			dw_mci_set_ios(slot->mmc, &slot->mmc->ios);
-		/* Force setup bus to guarantee available clock output 
-*/
+		/* Force setup bus to guarantee available clock output */
 		dw_mci_setup_bus(slot, true);
 
 	}
-- 
2.10.2

